alica_dummy_proxy/src/communication/AlicaDummyCommunication.cpp

```diff
@@ -213,8 +213,13 @@ typedef struct Queues
 CommModuleContainer AlicaDummyCommunication::s_modContainer;
 Queues AlicaDummyCommunication::s_qctx(s_modContainer);
 
-AlicaDummyCommunication::AlicaDummyCommunication(alica::AlicaEngine* ae)
-        : alica::IAlicaCommunication(ae)
+AlicaDummyCommunication::AlicaDummyCommunication(std::function<void(std::shared_ptr<alica::SyncTalk>)> onSyncTalkHandler,
+        std::function<void(std::shared_ptr<alica::SyncReady>)> onSyncReadyHandler,
+        std::function<void(const alica::AllocationAuthorityInfo&)> incomingAuthorityMessageHandler,
+        std::function<void(std::shared_ptr<alica::PlanTreeInfo>)> planTreeInfohandler, std::function<void(const alica::SolverResult&)> onSolverResultHandler,
+        std::function<void(const alica::AgentQuery&)> agentQueryHandler, std::function<void(const alica::AgentAnnouncement&)> agentAnnouncementHandler)
+        : alica::IAlicaCommunication(onSyncTalkHandler, onSyncReadyHandler, incomingAuthorityMessageHandler, planTreeInfohandler, onSolverResultHandler,
```

Comment:
In IAlicaCommunication you can define a struct AlicaCommunicationCallbacks with all these callbacks as members. That way you can make the constructors less verbose and just take the struct reference in constructors.

Reply:
done